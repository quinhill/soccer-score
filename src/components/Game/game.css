p {
  color: white;
  margin: 10px;
}

.game {
  width: 80%;
  display: grid;
  grid-template-columns: 35% 30% 35%;
}