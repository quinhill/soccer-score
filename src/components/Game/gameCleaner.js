export const timeCleaner = (time) => {
  let newTime = time.split('T')[1].split('+')[0]
  newTime = newTime.split(':')
  if (newTime[0] <= 12) {
    newTime = `${newTime[0]}:${newTime[1]}AM`
  } else {
    newTime = `${newTime[0] - 12}:${newTime[1]}PM`
  }
  return newTime
}