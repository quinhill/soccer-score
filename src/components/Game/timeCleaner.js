export const cleanTime = (time) => {
  let newTime = time.split(':')
  if (newTime[0] < 12) {
    newTime = `${newTime[0]}:${newTime[1]}AM`
  } else {
    newTime = `${newTime[0] - 12}:${newTime[1]}PM`
  }
  return newTime
}