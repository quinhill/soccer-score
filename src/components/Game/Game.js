import React from 'react';
import './game.css';

export const Game = (props) => {

  const handleClick = (event) => {
    const id = event.target.value;
    props.fetchTeam(id)
  }

  return (
    <div className="game">
      <button 
        className="button team-one"
        onClick={handleClick}
        value={props.competitors[0].id}
      >
        {props.competitors[0].name}
      </button>
      <p className="time">{props.scheduled}</p>
      <button 
        className="button team-two"
        onClick={handleClick}
        value={props.competitors[1].id}
      >
        {props.competitors[1].name}
      </button>
    </div>
  )
}