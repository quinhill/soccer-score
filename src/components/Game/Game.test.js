import React from 'react';
import { Game } from './Game';
import { shallow, mount } from 'enzyme';

describe('Game', () => {

  describe('handleClick', () => {

    let wrapper;
    let mockFetchTeam;
    let mockProps;

    beforeEach(() => {
      mockFetchTeam = jest.fn();
      mockProps = {
        competitors: [
          {id: 'sr:competitor:2900'},
          {id: 'sr:competitor:1000'}
        ]
      }
      
      wrapper = mount(<Game
        fetchTeam={mockFetchTeam}
        {...mockProps}
        />);
      })
      
    it('should call fetchTeam through props', () => {
      wrapper.find('.team-one').simulate('click')
        
      expect(mockFetchTeam).toHaveBeenCalled()
    })

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

})