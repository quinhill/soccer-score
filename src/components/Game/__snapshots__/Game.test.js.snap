// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game handleClick should match snapshot 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <Unknown
    competitors={
        Array [
            Object {
              "id": "sr:competitor:2900",
            },
            Object {
              "id": "sr:competitor:1000",
            },
          ]
    }
    fetchTeam={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "competitors": Array [
        Object {
          "id": "sr:competitor:2900",
        },
        Object {
          "id": "sr:competitor:1000",
        },
      ],
      "fetchTeam": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="game"
>
        <button
                class="button team-one"
                value="sr:competitor:2900"
        />
        <p
                class="time"
        />
        <button
                class="button team-two"
                value="sr:competitor:1000"
        />
</div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="button team-one"
            onClick={[Function]}
            value="sr:competitor:2900"
/>,
          <p
            className="time"
/>,
          <button
            className="button team-two"
            onClick={[Function]}
            value="sr:competitor:1000"
/>,
        ],
        "className": "game",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <button
            class="button team-one"
            value="sr:competitor:2900"
/>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "button team-one",
            "onClick": [Function],
            "value": "sr:competitor:2900",
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "button",
        },
        Object {
          "instance": <p
            class="time"
/>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "time",
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": <button
            class="button team-two"
            value="sr:competitor:1000"
/>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "button team-two",
            "onClick": [Function],
            "value": "sr:competitor:1000",
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "competitors": Array [
          Object {
            "id": "sr:competitor:2900",
          },
          Object {
            "id": "sr:competitor:1000",
          },
        ],
        "fetchTeam": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="game"
>
          <button
                    class="button team-one"
                    value="sr:competitor:2900"
          />
          <p
                    class="time"
          />
          <button
                    class="button team-two"
                    value="sr:competitor:1000"
          />
</div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="button team-one"
              onClick={[Function]}
              value="sr:competitor:2900"
/>,
            <p
              className="time"
/>,
            <button
              className="button team-two"
              onClick={[Function]}
              value="sr:competitor:1000"
/>,
          ],
          "className": "game",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <button
              class="button team-one"
              value="sr:competitor:2900"
/>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "button team-one",
              "onClick": [Function],
              "value": "sr:competitor:2900",
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "button",
          },
          Object {
            "instance": <p
              class="time"
/>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "time",
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": <button
              class="button team-two"
              value="sr:competitor:1000"
/>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "button team-two",
              "onClick": [Function],
              "value": "sr:competitor:1000",
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
